go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -64,6 +64,8 @@ type CommonNetworkControllerInfo struct {
 
 	// Northbound database zone name to which this Controller is connected to - aka local zone
 	zone string
+
+	nodeName string
```

Comment:
Not sure this fits the non IC model, or the multi-node zone IC model.

We are not bound to support those so maybe not a major issue.

But I was under the same situation for layer2.

What I did was just lookup through all the nodes and pick the first Id I would find for a given network ID on any node. 

The fact that we are annotating the network IDs on the nodes is because of the lack of a better place but all the nodes should have the same ID for the same network and it might not be relevant from which node we pick that up really.

You can pick that up from my final multi-honing changes if interested
https://github.com/jcaamano/ovn-kubernetes/blob/ic-l2-transit/go-controller/pkg/ovn/zone_interconnect/zone_ic_handler.go

Reply:
OK I'll just wait for your thing to merge then leverage it